kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/plan/names.kt

```diff
@@ -1,210 +1,213 @@
+ * a display name - that may have been customized by the user or the framework - for reporting purposes.
+ *
+ * This class is a long term replacement for [DescriptionName].
+ */
+@ExperimentalKotest
+sealed class NodeName {
+
+   companion object {
+
+      /**
+       * Returns a [SpecName] for the given class.
+       *
+       * For name, the following is used, in order of preference:
+       * - The fully qualified name of the class (JVM)
+       * - The simple name of the class (all platforms)
+       *
+       * For display name, the following are used, in order of preference:
+       * - If the spec has been annotated with @DisplayName and annotations are supported on that platform (JVM)
+       * - The fully qualified name of the class (JVM)
+       * - The simple name of the class
+       * - The [toString] on the [kclass] instance.
+       *
+       * Note: This name must be globally unique. Two specs, even in different packages,
+       * cannot share the same names, so if @DisplayName is used, developers must ensure it does not
+       * clash with another spec.
+       */
+      fun fromSpec(kclass: KClass<*>): SpecName {
+         val name = kclass.bestName()
+         val displayName = kclass.annotation<DisplayNameAnno>()?.name ?: kclass.simpleName ?: this.toString()
+         return SpecName(name, displayName)
+      }
+
+      @Deprecated("For compatibility until Descriptions are removed")
+      internal fun fromTestName(name: DescriptionName.TestName): TestName {
+         return TestName(
+            name = name.name,
+            displayName = name.displayName,
+            prefix = null,
+            suffix = null,
+            focus = name.focus,
+            bang = name.bang,
+         )
+      }
+
+      /**
+       * Parses a [TestName] correctly handling focus, bang, prefix and suffix.
+       * If a prefix is specified the focus/bang is moved to before the prefix in the display name.
+       *
+       * This means the user writes when("!disable") and the platform invokes ("when", "!disable")
+       * and ends up with !when disable, so that it is correctly parsed by the test runtime.
+       *
+       * @param prefix optional prefix that some specs may specify, such as "Given:"
+       * @param name the user supplied name for the test
+       * @param suffix optional suffix that some specs may specify such as "should"
+
+       * @param testNameCase a [TestNameCase] parameter to adjust the capitalisation of the display name
+       * @param includeAffixesInDisplayName if true then the prefix and/or suffix will be included in the display name.
+       * if false, they will not be included in the display name. If null, the default from config will be used.
+       */
+      @ExperimentalKotest
+      @PublishedApi
+      internal fun parseTestName(
+         prefix: String?,
+         name: String,
+         suffix: String?,
+         testNameCase: TestNameCase,
+         removeTestNameWhiteSpec: Boolean,
```

Comment:
Is this a typo? White space?

Reply:
Well its used as part of the create name method, a copy n pasta